### PR TITLE
[5.2] Add PubNub event broadcasting

### PR DESCRIPTION
**Pull request by cwt137**
This is a patch for issue #8684

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
This needs to be added to the suggested dependencies list too.

File: src/Illuminate/Broadcasting/Broadcasters/PubnubBroadcaster.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    {
+        $this->pubnub = $pubnub;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function broadcast(array $channels, $event, array $payload = array())
+    {
+
```
Comment (GrahamCampbell) [inline src/Illuminate/Broadcasting/Broadcasters/PubnubBroadcaster.php]: no blank line plz

File: composer.json [hunk 2]
```diff
@@ -108,7 +108,8 @@
         "league/flysystem-aws-s3-v2": "Required to use the Flysystem S3 driver (~1.0).",
         "league/flysystem-rackspace": "Required to use the Flysystem Rackspace driver (~1.0).",
         "pda/pheanstalk": "Required to use the beanstalk queue driver (~3.0).",
-        "predis/predis": "Required to use the redis cache and queue drivers (~1.0)."
+        "predis/predis": "Required to use the redis cache and queue drivers (~1.0).",
+        "pubnub/pubnub": "Required to use the PubNub event broadcast driver (dev-master)."
```
Comment (GrahamCampbell) [inline composer.json]: needs to go into both composer.json files ;)

Comment (GrahamCampbell) [inline composer.json]: Also, NOT dev-master!

Comment (GrahamCampbell) [inline composer.json]: ~3.7 would probably be a good idea here

**Comment by taylorotwell** [comment]
Thanks for the PR, will try to play with this service this week.

File: composer.json [hunk 3]
```diff
@@ -108,7 +108,8 @@
         "league/flysystem-aws-s3-v2": "Required to use the Flysystem S3 driver (~1.0).",
         "league/flysystem-rackspace": "Required to use the Flysystem Rackspace driver (~1.0).",
         "pda/pheanstalk": "Required to use the beanstalk queue driver (~3.0).",
-        "predis/predis": "Required to use the redis cache and queue drivers (~1.0)."
+        "predis/predis": "Required to use the redis cache and queue drivers (~1.0).",
+        "pubnub/pubnub": "Required to use the PubNub event broadcast driver (~3.7)."
```
Comment (GrahamCampbell) [inline composer.json]: You need to add this in BOTH composer.json files please.

Comment (GrahamCampbell) [inline composer.json]: Ping @cwt137 You need to fix merge conflicts, and you need to add and entry to https://github.com/laravel/framework/blob/master/src/Illuminate/Broadcasting/composer.json#L32.

**Comment by cwt137** [comment]
@taylorotwell did you have a chance to test it?

**Comment by GrahamCampbell** [comment]
Needs squashing to one commit.

**Comment by cwt137** [comment]
done

File: composer.json [hunk 4]
```diff
@@ -109,7 +109,8 @@
         "league/flysystem-rackspace": "Required to use the Flysystem Rackspace driver (~1.0).",
         "pda/pheanstalk": "Required to use the beanstalk queue driver (~3.0).",
         "predis/predis": "Required to use the redis cache and queue drivers (~1.0).",
-        "pusher/pusher-php-server": "Required to use the Pusher broadcast driver (~2.0)."
+        "pusher/pusher-php-server": "Required to use the Pusher broadcast driver (~2.0).",
+        "pubnub/pubnub": "Required to use the PubNub event broadcast driver (~3.7)."
```
Comment (GrahamCampbell) [inline composer.json]: You need to add this to **both** composer.json files.

**Comment by taylorotwell** [comment]
Going to hold off on more drivers for now, but please feel free to release this as a package.

**Comment by cwt137** [comment]
Done. https://packagist.org/packages/cwt137/laravel-pubnub-broadcast-driver

**Pull request closed without merging**